apps/cowswap-frontend/src/legacy/state/swap/reducer.ts

```diff
@@ -199,7 +199,7 @@ function getEthFlowOverridesOnSelect(
 ): Pick<SwapState, 'independentField' | 'typedValue'> {
   if (
     inputCurrencyId?.toUpperCase() ===
-    NATIVE_CURRENCY_BUY_TOKEN[state.chainId || ChainId.MAINNET]?.symbol?.toUpperCase()
+    NATIVE_CURRENCY_BUY_TOKEN[(state.chainId as ChainId) || ChainId.MAINNET]?.symbol?.toUpperCase()
```

Comment:
shouldn't `state.chainId` be of type ChainId in the first place? or this is a number to allow users to be in any other network? 
and if so, is the casting safe?

Reply:
It definitely must. I'm afraid of changing `SwapState`, it will cause of lots of new changes